odoo/addons/base/models/res_partner.py

```diff
@@ -606,15 +606,16 @@ def _commercial_sync_from_company(self):
             self.write(sync_vals)
             self._commercial_sync_to_children()
 
-    def _commercial_sync_to_children(self):
+    def _commercial_sync_to_children(self, fields_to_sync=None):
         """ Handle sync of commercial fields to descendants """
         commercial_partner = self.commercial_partner_id
-        sync_vals = commercial_partner._update_fields_values(self._commercial_fields())
+        if fields_to_sync is None:
+            fields_to_sync = self._commercial_fields()
+        sync_vals = commercial_partner._update_fields_values(fields_to_sync)
         sync_children = self.child_ids.filtered(lambda c: not c.is_company)
         for child in sync_children:
-            child._commercial_sync_to_children()
+            child._commercial_sync_to_children(fields_to_sync)
         res = sync_children.write(sync_vals)
-        sync_children._compute_commercial_partner()
```

Comment:
I think this was called to compute the commercial partner to flush `is_company` and `commercial_partner_id` fields.
You might want to flush `commercial_partner_id` of self._child_ids before applying this.

Reply:
This line dated from 2017 and was used to fix a bug on  `commercial_partner_id` compute.

I cannot reproduce the bug described in https://github.com/odoo/odoo/commit/0d68acff8e4436e55676a5016e0496b7c3aa35aa.
It was probably fixed with the new `recursive=True` attribute.